### DISCUSSION TITLE
doxygen problem

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#168] [Design & Development]
This isn't a big issue but I think there is some setting wrong in the doxygen configuration file.  It seems the documentation being generated is only creating doc pages for C++ classes.  We have a bunch of C++ functions that are not currently documented.  Some are important like those with python wrappers for windowing data and computing amplitudes - at least.   

I know doxygen fairly well.  If you can give me a hint where that configuration file is I might be able to fix it.

**Comment by wangyinz** [comment]
There you go: https://github.com/wangyinz/mspass/blob/master/docs/source/Doxyfile

I think we haven't touched it for a while so it is not surprising that those later added functions are missing.

**Reply by pavlis** [reply]
Thanks.  I looking through this I realized I couldn't really test this at this time since I still have no complete, functional system to build the sphinx documentation.  Guess I need to fix that, but for this case I think I can just make some suggestions:

1.  line 917:  RECURSIVE should be yes.   We are not seeing deconvolution now for that reason.  I would suggest we try just setting this yes and also setting INPUT to just cxx/src and cxx/include
2. I think it would useful to turn on (set YES) REFERENCES_RELATION (1073) and REFERENCES_LINK_SOURCES (1067).  
3. I recommend turning on (set YES) USE_MATHJAX.   I think there are already a few latex formulas i the docs, but this makes it more feasible to embed math formulas.
4. line 1054:  probably should set INLINE_SOURCES to YES.  I think this might cause template functions to be excluded otherwise. 

Unfortunately, none of those are clearly responsible for the problem I see.  1 is the only one that has much hope of fixing this problem.
